hack/tools/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/sigstore/hack/tools
 
-go 1.16
+go 1.17
```

Comment:
need to run go mod tidy in this directory as well

Reply:
Got it, I forgot how much handholding go dependencies need after working with cargo.